### DISCUSSION TITLE
Transition from manual to automatic secrets encryption.

### DISCUSSION THREAD

**Discussion by xhejtman** [rancher/rke2#7330] [Q&A]
Hello,

we run cluster for several years, beginning with versions below 1.21 that dit not have secrets-encryption options. In that days, we created encryption-config.json manually and passed it to kube-api server as an options. I would like to migrate to rke2 secrets-encryption, but I did not find how can I pass initial encryption key to the rke2 as rke2 seems to be generating own encryptions keys. Is there any way how to migrate the current encryption into rke2 encryption?

**Comment by dereknola** [comment]
There is no easy way to migrate from using manual encryption-config.json to the RKE2 secret-encrypt system. We never designed it with migration in mind, so all the code is written as a closed system.
